doc/book/cookbook/setting-module-specific-layouts.md

```diff
@@ -0,0 +1,154 @@
+use Zend\Filter\Word\CamelCaseToDash;
+use Zend\Mvc\MvcEvent;
+use Zend\View\Resolver\TemplateMapResolver;
+
+class LayoutListener extends AbstractListenerAggregate
+{
+    /** @var TemplateMapResolver */
+    private $templateMapResolver;
+
+    /** @var FilterInterface */
+    private $filter;
+
+    public function __construct(TemplateMapResolver $templateMapResolver)
+    {
+        $this->templateMapResolver = $templateMapResolver;
+        $this->filter              = (new FilterChain())
+            ->attach(new CamelCaseToDash())
+            ->attach(new StringToLower());
+    }
+
+    public function attach(EventManagerInterface $events, $priority = 1)
+    {
+        $this->listeners[] = $events->attach(
+            MvcEvent::EVENT_RENDER,
+            [$this, 'setLayout']
+        );
+    }
+
+    public function setLayout(MvcEvent $event) : void
+    {
+        // Get and check the route match object
+        $routeMatch = $event->getRouteMatch();
+        if (! $routeMatch) {
+            return;
+        }
+
+        // Get and check the parameter for current controller
+        $controller = $routeMatch->getParam('controller');
+        if (! $controller) {
+            return;
+        }
+    
+        // Extract module name
+        $module = substr($controller, 0, strpos($controller, '\\'));
+
+        // Convert the module name from camel case to a lower string with dashes
+        $name = 'layout/' . $this->filter->filter($module);
+
+        // Has the resolver an entry / layout with the given name?
+        if (! $this->templateMapResolver->has($name)) {
+            return;
+        }
+
+        // Get root view model
+        $layoutViewModel = $event->getViewModel();
+
+        // Change template
+        $layoutViewModel->setTemplate($name);
+    }
+}
+```
+
+## Register Listener
+
+Extend the module class to register the listener, e.g.
+`module/Admin/Module.php`:
+
+```php
+namespace Admin;
+
+use Application\Listener\LayoutListener;
+use Zend\Mvc\MvcEvent;
+use Zend\View\Resolver\TemplateMapResolver;
+
+class Module
+{
+    public function onBootstrap(MvcEvent $event) : void
+    {
+        $application = $event->getApplication();
+
+        /** @var TemplateMapResolver $templateMapResolver */
+        $templateMapResolver = $application->getServiceManager()->get(
+            'ViewTemplateMapResolver'
+        );
+        
+        // Create and register layout listener
+        $listener = new LayoutListener($templateMapResolver);
+        $listener->attach($application->getEventManager());
+    }
+
+    // …
+}
+```
+
+> More informations on registering module-specific listeners can be found in the 
+> [documentation of zend-mvc](https://docs.zendframework.com/zend-mvc/examples/#registering-module-specific-listeners).
+
+## Add Template Scripts to the Configuration
+
+Extend the configuration of a module to add the specific layout script, e.g.
+`module/Admin/config/module.config.php`:
```

Comment:
As we refer this file here, I would add also this into Module.php snipped above:

```
    public function getConfig()
    {
        return include __DIR__ . '/../config/module.config.php';
    }
```

Just for completeness.

Reply:
Unfortunately that does not fit the description above:

> Extend the module class to register the listener…

All code examples are based on an existing zend-mvc application. (See the introduction of this cookbook recipe.)